src/NumericalAlgorithms/Spectral/Projection.cpp

```diff
@@ -19,36 +19,38 @@
 
 namespace Spectral {
 
-std::ostream& operator<<(std::ostream& os, MortarSize mortar_size) noexcept {
+std::ostream& operator<<(std::ostream& os, ChildSize mortar_size) noexcept {
   switch (mortar_size) {
-    case MortarSize::Full:
+    case ChildSize::Full:
       return os << "Full";
-    case MortarSize::UpperHalf:
+    case ChildSize::UpperHalf:
       return os << "UpperHalf";
-    case MortarSize::LowerHalf:
+    case ChildSize::LowerHalf:
       return os << "LowerHalf";
     default:
-      ERROR("Invalid MortarSize");
+      ERROR(
+          "Invalid ChildSize. Expected one of: 'Full', 'UpperHalf', "
+          "'LowerHalf'");
   }
 }
 
-const Matrix& projection_matrix_mortar_to_element(
-    const MortarSize size, const Mesh<1>& element_mesh,
-    const Mesh<1>& mortar_mesh) noexcept {
-  ASSERT(element_mesh.basis(0) == Basis::Legendre and
-             mortar_mesh.basis(0) == Basis::Legendre,
+const Matrix& projection_matrix_child_to_parent(
```

Comment:
Do you think naming these function `restriction_matrix` and `prolongation_matrix` would be clearer than the current names?